core/src/main/java/org/testcontainers/images/builder/ImageFromDockerfile.java

```diff
@@ -46,6 +46,9 @@
 
     static {
         Runtime.getRuntime().addShutdownHook(new Thread(DockerClientFactory.TESTCONTAINERS_THREAD_GROUP, () -> {
+            if (imagesToDelete.isEmpty()) {
```

Comment:
unrelated to the main change, but I noticed that we register a shutdown hook that asks for a client even if the list is empty and no actions has to be performed, which means that we will create & resolve a docker client even if we run a simple unit test :)

Reply:
Will also be fixed once https://github.com/testcontainers/testcontainers-java/pull/1745 is merged